polaris/dataset/_dataset.py

```diff
@@ -185,69 +183,24 @@ def _compute_checksum(self):
         hash_fn.update(table_hash)
 
         # If the Zarr archive exists, we hash its contents too.
-        zarr_md5sum_manifest = None
-        if self.zarr_root_path is not None:
-            zarr_hash, zarr_md5sum_manifest = compute_zarr_checksum(self.zarr_root_path)
+        if self.uses_zarr:
+            zarr_hash, self._zarr_md5sum_manifest = compute_zarr_checksum(self.zarr_root_path)
             hash_fn.update(zarr_hash.encode())
 
         checksum = hash_fn.hexdigest()
-        return checksum, zarr_md5sum_manifest
-
-    def verify_checksum(self, md5sum: Optional[str] = None):
-        """
-        Recomputes the checksum and verifies whether it matches the stored checksum.
-
-        Warning: Slow operation
-            This operation can be slow for large datasets.
-
-        Info: Only works for locally stored datasets
-            The checksum verification only works for datasets that are stored locally in its entirety.
-            We don't have to verify the checksum for datasets stored on the Hub, as the Hub will do this on upload.
-            And if you're streaming the data from the Hub, we will check the checksum of each chunk on download.
-        """
-        if md5sum is None:
-            md5sum = self._md5sum
-        if md5sum is None:
-            logger.warning(
-                "No checksum to verify against. Specify either the md5sum parameter or "
-                "store the checksum in the dataset.md5sum attribute."
-            )
-            return
-
-        # Temporarily reset
-        # Calling self.md5sum will recompute the checksum and set it again
-        self._md5sum = None
-        if self.md5sum != md5sum:
-            raise PolarisChecksumError(
-                f"The specified checksum {md5sum} does not match the computed checksum {self.md5sum}"
-            )
-
-    @computed_field
-    @property
-    def md5sum(self) -> str:
-        """Lazily compute the checksum once needed."""
-        if not self.has_md5sum:
-            self._md5sum, self._zarr_md5sum_manifest = self._compute_checksum()
-        return self._md5sum
-
-    @md5sum.setter
-    def md5sum(self, value: str):
-        """Set the checksum."""
-        if len(value) != 32 or not all(c in "0123456789abcdef" for c in value):
-            raise ValueError("The checksum should be a 32-character long MD5 hash.")
-        self._md5sum = value
-
-    @property
-    def has_md5sum(self) -> bool:
-        """Whether the md5sum for this class has been computed and stored."""
-        return self._md5sum is not None
+        return checksum
```

Comment:
Should this return the manifest, too? Seems like either that or we should update the abstract class description (currently says "Compute the checksum and return it along with manifest.")

Reply:
That's a mistake in the docs! Good catch!